frontend/src/plugin/registry.tsx

```diff
@@ -384,3 +389,18 @@ export function registerAppLogo(logo: AppLogoType) {
 export function registerClusterChooser(chooser: ClusterChooserType) {
   store.dispatch(setClusterChooserButtonComponent(chooser));
 }
+
+/**
+ * Override headlamp helper functions
+ * @param override - The functions to override
+ *
+ * @example
+ * ```ts
+ * registerFunctionsToOverride({
+ *  getClusterName: () => 'my-cluster',
+ *  getToken: () => 'token'
+ * });
+ */
+export function registerFunctionsToOverride(override: FunctionsToOverride) {
+  store.dispatch(setFunctionsToOverride(override));
+}
```

Comment:
What I meant in my previous review is that we shouldn't export this function since we don't have a clear plan yet to use it. So we can keep the internal redux logic you added, but let's export a `registerSetTokenFunction()` and similar for `getToken`.

Later if we have a clear plan or many functions to override, we can bring back this `registerFunctionsToOverride` function easily (since we'll keep the redux internals).